lib/layer.js

```diff
@@ -48,6 +48,31 @@ function Layer(path, options, fn) {
 }
 
 /**
+ * Handle any request for the layer.
+ *
+ * @api private
+ */
+
+Layer.prototype._handle_any = function _handle_any(/* ...args, next */) {
+  var self = this
+  var next = arguments[arguments.length - 1]
+  var handle = this.handle
+  var result
+  var hasNextBeenCalled = function () { return next._currentLayer === self }
+
+  try {
+    result = handle.apply(undefined, arguments)
```

Comment:
Hi @calebmer , your change does not fix it, because the `this` in your code is still executing from within the strict code in the router source code, which is not where the `this` varies at.

Your test above is also very different from the test I gave, as the `this` is always being executed in sloppy mode, not in the caller's defined code. Please refer to my example for what I'm talking about :)